src/Pages/Portofolio.jsx

```diff
@@ -0,0 +1,390 @@
+                    >
+                      <TableCell
+                        component="th"
+                        id={labelId}
+                        scope="asset"
+                        padding="none"
+                      >
+                        {assetSymbol}
+                      </TableCell>
+                      <TableCell align="right"> $ {avgPrice}</TableCell>
+                      <TableCell align="right">{quantity}</TableCell>
+                      <TableCell align="right">$ {currentPrices.c}</TableCell>
+                      <TableCell align="right"> $ {profit}</TableCell>
+                      <TableCell align="right">
+                        <Stack
+                          component="form"
+                          sx={{ width: "10ch" }}
+                          spacing={1}
+                        >
+                          <div className="two-columns-grid">
+                            <input
+                              className="quantity-input-field"
+                              type="number"
+                              min="0"
+                              step="1"
+                              onChange={handleTrackQuantity}
+                            />
+                            <Button size="small" variant="contained"
+                             name="sell"
+                             onClick={(e) => handleSellQuantity(e, asset)}
+                            >
+                              Sell
+                            </Button>
+                          </div>
+                          <Button
+                            className="sell-all-btn"
+                            size="small"
+                            variant="contained"
+                            name="all"
+                            onClick={(e) => handleSellQuantity(e, asset)}
+                          >
+                            Sell All
+                          </Button>
+                        </Stack>
+                      </TableCell>
+                    </TableRow>
+                  );
+                })}
+              {emptyRows > 0 && (
+                <TableRow // Dense padding toggle functionality
+                  style={{
+                    height: (dense ? 33 : 53) * emptyRows,
+                  }}
+                >
+                  <TableCell colSpan={6} />
+                </TableRow>
+              )}
+            </TableBody>
+          </Table>
+        </TableContainer>
+        <TablePagination
+          rowsPerPageOptions={[5, 10, 25]}
+          component="div"
+          count={portfolio.length}
+          rowsPerPage={rowsPerPage}
+          page={page}
+          onPageChange={handleChangePage}
+          onRowsPerPageChange={handleChangeRowsPerPage}
+        />
+      </Paper>
+      <FormControlLabel
+        control={<Switch checked={dense} onChange={handleChangeDense} />}
+        label="Dense padding"
+      />
+    </Box>
+  );
+}
```

Comment:
Overall this is good, but there are a lot of things in here that could be their own components in their own files, e.g. `EnhancedTableToolbar`